python/hsfs/constructor/query.py

```diff
@@ -237,6 +238,7 @@ def as_of(self, wallclock_time=None, exclude_until=None):
         # Returns
             `Query`. The query object with the applied time travel condition.
         """
+        self._is_time_travel = True
```

Comment:
setting `_is_time_travel` should be move at the end of the function. Otherwise it would set _it_time_travel to true, even if, e.g., `until.convert_event_time_to_timestamp` fails and time travel never gets activated.

Reply:
Also, if as_of gets called with both arguments `None`, we should not set `_is_time_travel=True`